File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash;
        vector<int> answer;
        
        for(int i = 0;i<nums.size();i++){
            if(hash.find(target-nums[i]) != hash.end()){
                answer.push_back(hash[target-nums[i]]);
                answer.push_back(i);
            }
            hash[nums[i]] = i;
        }
        return answer;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        hashSet = defaultdict()

        for index , num in enumerate(nums) :
            if target - num in hashSet :
                return [hashSet[target-num],index]

            hashSet[num] = index

        return [0,0]